### ISSUE TITLE
m-btn-group used outside of buttons

### ISSUE THREAD

**Issue by anselmbradford** [cfpb/capital-framework#946]
## Current behavior

https://cfpb.github.io/capital-framework/components/cf-buttons/#button-links shows a pattern for making buttons look like links. These can be combined with button groups to provide a button and a link together https://cfpb.github.io/capital-framework/components/cf-buttons/#button-group

However, in the wild, this allows buttons and links to be grouped together, such as this markup on the [blog filter](https://www.consumerfinance.gov/about-us/blog/)

```
<div class="content-l_col
            content-l_col-1
            m-btn-group">
    <input class="a-btn" type="submit" value="Apply filters">
    <a class="a-btn a-btn__link a-btn__warning" href="/about-us/blog/">
         Clear filters
    </a>
</div>
```

## Expected behavior

- The CF patterns should account for button groups that contain actual links.

**Comment by Scotchester** [comment]
I'm not sure I understand what this issue specifically wants. More examples in the docs showing button groups using non-`button` elements?

**Comment by anselmbradford** [comment]
> I'm not sure I understand what this issue specifically wants. More examples in the docs showing button groups using non-button elements?

Yeah, if the pattern in https://www.consumerfinance.gov/blog is fine, we should codify it. Like should the link be `a-btn` or `a-link` inside a `m-btn-group`?
